src/plots/cartesian/set_convert.js

```diff
@@ -152,10 +152,8 @@ module.exports = function setConvert(ax, fullLayout) {
         if(ax._categoriesMap) {
             index = ax._categoriesMap[v];
             if(index !== undefined) return index;
-        } else {
-            index = ax._categories.indexOf(v);
-            if(index !== -1) return index;
```

Comment:
Right now no functions will fallback to use ``indexOf`` on ``_categories`` anymore.

Currently I made sure ``_categoriesMap`` is always populated accordingly when ``_categories`` is created or initialized with some items for the current code. But this is not very robust way. Some other contributors could change ``_categories``  without updating ``_categoriesMap`` in the future. And it will not be easy for them to debug the unit test and find out the reason behind breaks.